starknet_py/net/account/account.py

```diff
@@ -273,6 +274,11 @@ async def sign_declare_transaction(
                 version=2,
             )
         else:
+            if _is_sierra_contract(json.loads(compiled_contract)):
+                raise ValueError(
+                    "Argument compiled_class_hash is required when using sierra compiled_contract."
+                )
```

Comment:
Could we change it to
```
if is_sierra_contract:
   if compiled_class_hash is None: raise
   declare_tx = declare v2
else:
   declare_tx = declare v1
```

Reply:
We'd have then have handle `compiled_class_hash` in the else branch. Because it should not be provided with declare v1.

Reply:
Could we at least refactor this function a little? it got big and more complicated